### PR TITLE
Surface Window data struct & heat balance routines refactoring

### PR DESCRIPTION
**Pull request by xuanluo113**
Pull request overview
---------------------
This branch includes refactoring of the HB related data structures and scope reduction in `InitSolarHeatGains`. In detail, the changes include:
- Refactoring – naming convention (var name prefix)
e.g. Surf (for all surfaces), SurfWin, SurfOpaq 
- Reduce the scope of function level local variables.
e.g. Surface(SurfNum).FrameDivider > 0
- Reduce scope of loops (Separating into shading loops, opaq surface loops and window surface loops)
- Reduce redundant branches
e.g. CalcSolRefl
- Move branches out of the loop applying MCCF
e.g. TDD diffusers and TDD shelves

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [x] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [x] Perform a Code Review on GitHub
 - [x] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [x] CI status: all green or justified
 - [x] Check that performance is not impacted (CI Linux results include performance check)
 - [x] Run Unit Test(s) locally
 - [x] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/DataGlobals.cc
Comment (mjwitte) [inline src/EnergyPlus/DataGlobals.cc]: Did you intend to keep this here?

File: src/EnergyPlus/DataHeatBalSurface.cc
Comment (mjwitte) [inline src/EnergyPlus/DataHeatBalSurface.cc]: These (and others) are really enclosure variables, not zone, indexed on `enclosureNum` (or similar). Also, solar and radiant enclosures can be different. Seems like we should use different prefixes for enclosures? `SolEncl...` `RadEncl...` ?

File: src/EnergyPlus/DataSurfaces.cc
Comment (mjwitte) [inline src/EnergyPlus/DataSurfaces.cc]: Minor - there's an extra space here (and a few other places below).

File: src/EnergyPlus/HeatBalanceSurfaceManager.cc
Comment (mjwitte) [inline src/EnergyPlus/HeatBalanceSurfaceManager.cc]: Did you intend to keep this and related code? Should be put some ##ifdefs around it?

File: src/EnergyPlus/HeatBalanceSurfaceManager.cc
Comment (mjwitte) [inline src/EnergyPlus/HeatBalanceSurfaceManager.cc:2719]: Since all shading surfaces are at the top of the surface list, we should be able to add a new variable for `TotShadingSurfaces` (or `ShadingSurfaceLast` or similar name) and loop here from 1 to `TotShadingSurfaces`.

File: src/EnergyPlus/HeatBalanceSurfaceManager.cc
Comment (mjwitte) [inline src/EnergyPlus/HeatBalanceSurfaceManager.cc]: I think this line can be deleted.

File: src/EnergyPlus/HeatBalanceSurfaceManager.cc
Comment (mjwitte) [inline src/EnergyPlus/HeatBalanceSurfaceManager.cc:3131]: There are many places in this file that show formatting changes like this with no substantive code change. When I apply clang formatting to this line it goes back to a single line.

File: src/EnergyPlus/DataHeatBalSurface.cc
Comment (mjwitte) [inline src/EnergyPlus/DataHeatBalSurface.cc]: Minor, but names like this which already have `Surf` in them could simply be `SurfTempOut`.

File: third_party/ObjexxFCL/src/ObjexxFCL/Array1.hh
Comment (mjwitte) [inline third_party/ObjexxFCL/src/ObjexxFCL/Array1.hh]: ??

**Review by mjwitte — COMMENTED** [review]
@xuanluo113 I've merged in develop, resolved some minor conflicts and updated this branch.

Unit tests run locally all pass.

CI is looking good so far.  In general, this all looks good - variable naming clarifications and loop variable scope reductions. A few comments/questions:

**Comment by mjwitte** [comment]
@xuanluo113 Also, please edit the PR description at the top with a brief summary of the changes.

File: src/EnergyPlus/DataGlobals.cc
Comment (xuanluo113) [reply src/EnergyPlus/DataGlobals.cc]: No. Left by mistake. Just deleted these.

File: src/EnergyPlus/DataHeatBalSurface.cc
Comment (xuanluo113) [reply src/EnergyPlus/DataHeatBalSurface.cc]: I renamed `ZoneVMULT` and `QD's to use `SolEncl` prefix. I realized `QC`, `QDV` and `VCONV` are allocated but not used anymore, so I deleted them.

File: src/EnergyPlus/DataHeatBalSurface.cc
Comment (xuanluo113) [reply src/EnergyPlus/DataHeatBalSurface.cc]: I renamed this globally.

File: src/EnergyPlus/HeatBalanceSurfaceManager.cc
Comment (xuanluo113) [reply src/EnergyPlus/HeatBalanceSurfaceManager.cc:2719]: I'm actually working on this at the other branch, but haven't fully tested this piece yet. Do we want to pull that piece in this one? I was hoping this can go first so I'm not blocking Matt.

**Comment by xuanluo113** [comment]
@mjwitte Thanks for the review. I address them and updated the PR overview.

File: src/EnergyPlus/HeatBalanceSurfaceManager.cc
Comment (mjwitte) [reply src/EnergyPlus/HeatBalanceSurfaceManager.cc:2719]: This is ok here. It can wait for your next phase.

File: src/EnergyPlus/DataHeatBalSurface.cc
Comment (mjwitte) [reply src/EnergyPlus/DataHeatBalSurface.cc]: Looking at this now, I realize that this should be `EnclSol` to match the pattern of `SurfOpaq` `SurfWin` etc.  It's your call whether to change that in this branch or in your next phase. Sorry I didn't see that before.

Comment (xuanluo113) [reply src/EnergyPlus/DataHeatBalSurface.cc]: Makes sense. I renamed it at the latest commit.

**Review by mjwitte — APPROVED** [review]
@xuanluo113 Thanks for addressing my comments.

@Myoldmopar I'm finished reviewing this. Let me know if you want to review before merging.

**Comment by Myoldmopar** [comment]
Let it go in, I don't have anything further.

**Comment by xuanluo113** [comment]
@mjwitte @Myoldmopar thanks!

**Pull request merged**
